### PR TITLE
Merge sort dup subworkflow

### PR DESCRIPTION
**Pull request by guanqiaofeng**
To test subworkflow (under directory argo-modules/tests/subworkflows/icgc-argo-workflows/merg_sort_dup): 
```
nextflow run main.nf -entry test_merg_sort_dup --tools 'markdup' -profile docker -c ../../../config/nextflow.config
```

Expected outcome: 
```
N E X T F L O W  ~  version 23.10.1
Launching `main.nf` [kickass_hoover] DSL2 - revision: 378ebf63df
executor >  local (5)
[f5/1c4a48] process > test_merg_sort_dup:MERG_SORT_DUP:SAMTOOLS_MERGE (Study123.PatientXYZ.SampleA)                             [100%] 1 of 1 ✔
[21/fb6fd2] process > test_merg_sort_dup:MERG_SORT_DUP:BIOBAMBAM_BAMMARKDUPLICATES2 (Study123.PatientXYZ.SampleA.csort.markdup) [100%] 1 of 1 ✔
[87/c26ba2] process > test_merg_sort_dup:MERG_SORT_DUP:SAMTOOLS_INDEX (Study123.PatientXYZ.SampleA.csort.markdup)               [100%] 1 of 1 ✔
[e8/e6ca0b] process > test_merg_sort_dup:MERG_SORT_DUP:SAMTOOLS_CONVERT (Study123.PatientXYZ.SampleA.csort.markdup)             [100%] 1 of 1 ✔
[06/d0f9ff] process > test_merg_sort_dup:MERG_SORT_DUP:TAR (Study123.PatientXYZ.SampleA.RNA-seq.aln.cram.duplicates_metrics)    [100%] 1 of 1 ✔
WARN: There's no process matching config selector: SONG.*|SCORE.*
WARN: There's no process matching config selector: SONG_GET|SCORE_DOWNLOAD
WARN: There's no process matching config selector: SONG_SUBMIT|SONG_MANIFEST|SONG_PUBLISH|SCORE_UPLOAD
```

### CODE REVIEW

File: tests/subworkflows/icgc-argo-workflows/merg_sort_dup/test.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+- name: "merg_sort_dup"
+  command: nextflow run main.nf -entry test_merg_sort_dup --tools 'markdup' -profile docker -c ../../../config/nextflow.config
```
Comment (edsu7) [inline tests/subworkflows/icgc-argo-workflows/merg_sort_dup/test.yml:2]: Needs a default `--tools` value. 

Running the following :
`nextflow run main.nf -entry test_merg_sort_dup --tools '' -profile docker -c ../../../config/nextflow.config`
results in:
```
N E X T F L O W  ~  version 23.04.1
Launching `main.nf` [mad_banach] DSL2 - revision: 378ebf63df
Missing process or function split([,])

 -- Check script 'main.nf' at line: 47 or see '.nextflow.log' file for more details
```
`nextflow run main.nf -entry test_merg_sort_dup  -profile docker -c ../../../config/nextflow.config`
results in:
```
N E X T F L O W  ~  version 23.04.1
Launching `main.nf` [drunk_meninsky] DSL2 - revision: 378ebf63df
ERROR ~ Cannot invoke method split() on null object

 -- Check script '../../../../subworkflows/icgc-argo-workflows/merg_sort_dup/main.nf' at line: 107 or see '.nextflow.log' file for more details
```
But with:
`nextflow run main.nf -entry test_merg_sort_dup --tools ' ' -profile docker -c ../../../config/nextflow.config`
```
N E X T F L O W  ~  version 23.04.1
Launching `main.nf` [awesome_rubens] DSL2 - revision: 378ebf63df
executor >  local (3)
[78/e030a9] process > test_merg_sort_dup:MERG_SORT_DUP:SAMTOOLS_MERGE (Study123.PatientXYZ.SampleA) [100%] 1 of 1 ✔
[29/590c6c] process > test_merg_sort_dup:MERG_SORT_DUP:SAMTOOLS_INDEX (Study123.PatientXYZ.Sampl... [100%] 1 of 1 ✔
[03/bf5806] process > test_merg_sort_dup:MERG_SORT_DUP:SAMTOOLS_CONVERT (Study123.PatientXYZ.Sam... [100%] 1 of 1 ✔
```

Comment (edsu7) [inline tests/subworkflows/icgc-argo-workflows/merg_sort_dup/test.yml:2]: Also default should match other `test.yml`
https://github.com/icgc-argo-workflows/argo-modules/blob/3d0992ba3996eef2066f961304d03c1ed69d2174/tests/subworkflows/icgc-argo-workflows/song_score_upload/test.yml#L3
https://github.com/icgc-argo-workflows/argo-modules/blob/3d0992ba3996eef2066f961304d03c1ed69d2174/tests/modules/icgc-argo-workflows/checkinput/test.yml#L4

File: tests/subworkflows/icgc-argo-workflows/merg_sort_dup/test.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+- name: "merg_sort_dup"
+  command: nextflow run ./tests/subworkflows/icgc-argo-workflows/merg_sort_dup -entry test_merg_sort_dup --tools 'markdup' -profile docker -c ./tests/config/nextflow.config
```
Comment (guanqiaofeng) [inline tests/subworkflows/icgc-argo-workflows/merg_sort_dup/test.yml:2]: Addressed the comment of testing code style consistent with others

File: tests/subworkflows/icgc-argo-workflows/merg_sort_dup/main.nf [hunk 3]
```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+// Set default parameter values
+params.tools = ' '
```
Comment (guanqiaofeng) [inline tests/subworkflows/icgc-argo-workflows/merg_sort_dup/main.nf:6]: addressed the comment of setting default value for --tools

File: tests/subworkflows/icgc-argo-workflows/merg_sort_dup/test.yml (hunk 2 above)
Comment (edsu7) [reply tests/subworkflows/icgc-argo-workflows/merg_sort_dup/test.yml:2]: Add test case with default:
```
nextflow run ./tests/subworkflows/icgc-argo-workflows/merg_sort_dup -entry test_merg_sort_dup -profile docker -c ./tests/config/nextflow.config
```

**Review by edsu7 — CHANGES_REQUESTED** [review]

File: tests/subworkflows/icgc-argo-workflows/merg_sort_dup/test.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+- name: "merg_sort_dup_default"
+  command: nextflow run ./tests/subworkflows/icgc-argo-workflows/merg_sort_dup -entry test_merg_sort_dup -profile docker -c ./tests/config/nextflow.config
+    tags:
+    - "subworkflows"
+    - "subworkflows/merge_sort_dup"
+  files:
+    - path: "output/samtools_merge/Study123.PatientXYZ.SampleA.bam"
+    - path: "output/samtools_index/Study123.PatientXYZ.SampleA.csort.markdup.bam.bai"
+    - path: "output/samtools_convert/Study123.PatientXYZ.SampleA.csort.markdup.cram"
+    - path: "output/samtools_convert/Study123.PatientXYZ.SampleA.csort.markdup.cram.crai"
```
Comment (guanqiaofeng) [inline tests/subworkflows/icgc-argo-workflows/merg_sort_dup/test.yml:24]: Added test case with default.

File: tests/subworkflows/icgc-argo-workflows/merg_sort_dup/test.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+- name: "merg_sort_dup_default"
+  command: nextflow run ./tests/subworkflows/icgc-argo-workflows/merg_sort_dup -entry test_merg_sort_dup -profile docker -c ./tests/config/nextflow.config
+    tags:
+    - "subworkflows"
+    - "subworkflows/merge_sort_dup"
+  files:
+    - path: "output/samtools_merge/Study123.PatientXYZ.SampleA.bam"
+    - path: "output/samtools_index/Study123.PatientXYZ.SampleA.csort.markdup.bam.bai"
+    - path: "output/samtools_convert/Study123.PatientXYZ.SampleA.csort.markdup.cram"
```
Comment (edsu7) [inline tests/subworkflows/icgc-argo-workflows/merg_sort_dup/test.yml:23]: If no `markdup` step specified file should not be flagged with `markdup`

**Review by edsu7 — CHANGES_REQUESTED** [review]

**Review by edsu7 — APPROVED** [review]
LGTM

**Pull request merged**